src/m365/planner/commands/plan/plan-add.ts

```diff
@@ -0,0 +1,167 @@
+              cb();
+            }
+            else {
+              const requestOptions: any = {
+                url: endpointPlanner,
+                headers: {
+                  'accept': 'application/json;odata.metadata=none'
+                },
+                responseType: 'json',
+                data: {
+                  "owner": groups[0].id,
+                  "title": args.options.title
+                }
+              };
+              request
+                .post(requestOptions)
+                .then((res: any): void => {
+                  logger.log(res);
+                  cb();
+                }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+            }
+          }
+          else {
+            logger.log(`Owner group not found with id ${args.options.ownerGroupId}`);
+            cb();
+          }
+        }, (err: any): void => this.handleRejectedPromise(err, logger, cb));
+    }
+    else {
+      const ownerGroupNameFilter: string = args.options.ownerGroupName ? `?$filter=DisplayName eq '${encodeURIComponent(args.options.ownerGroupName).replace(/'/g, `''`)}'` : '';
+      this.getGroupsWithFilter(ownerGroupNameFilter)
+        .then((groups: any[]): void => {
+          if (groups && groups.length > 0) {
+            if (groups.length > 1) {
+              logger.log(`More than one groups found with name ${args.options.ownerGroupName}`);
+              cb();
+            }
+            else {
+              const requestOptions: any = {
+                url: endpointPlanner,
+                headers: {
+                  'accept': 'application/json;odata.metadata=none'
+                },
+                responseType: 'json',
+                data: {
+                  "owner": groups[0].id,
+                  "title": args.options.title
+                }
+              };
+              request
+                .post(requestOptions)
+                .then((res: any): void => {
+                  logger.log(res);
+                  cb();
+                }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+            }
+          }
+          else {
+            logger.log(`Owner group not found with name ${args.options.ownerGroupName}`);
+            cb();
+          }
+        }, (err: any): void => this.handleRejectedPromise(err, logger, cb));
+    }
+  }
+
+  private async getGroupsWithFilter(ownerGroupFilter: string): Promise<any[]> {
+    const endpoint: string = `${this.resource}/v1.0/groups${ownerGroupFilter}`;
+    const requestOptions: any = {
+      url: endpoint,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    try {
+      const response = await request
+        .get<{ value: any[]; }>(requestOptions);
+      return await Promise.resolve(response.value);
+    } catch (error) {
+      return await Promise.reject(error);
```

Comment:
Since we're returning the same error, what's the added value of wrapping the call in a `try..catch`?